test/easyconfigs/easyconfigs.py

```diff
@@ -1333,21 +1333,29 @@ def template_easyconfig_test(self, spec):
 
     # make sure all patch files are available
     specdir = os.path.dirname(spec)
+    basedir = os.path.dirname(os.path.dirname(specdir))
     specfn = os.path.basename(spec)
     for idx, patch in enumerate(ec['patches']):
-        if isinstance(patch, (tuple, list)):
-            patch = patch[0]
+        patch_dir = specdir
+        if isinstance(patch, str):
+            patch_name = patch
+        elif isinstance(patch, (tuple, list)):
+            patch_name = patch[0]
+        elif isinstance(patch, dict):
+            patch_name = patch['name']
+            if patch['alt_location']:
+                patch_dir = os.path.join(basedir, letter_dir_for(patch_name), patch_name)
```

Comment:
```suggestion
                patch_dir = os.path.join(basedir, letter_dir_for(patch['alt_location']), patch['alt_location'])
```
I think that is the correction needed.

Error from https://github.com/easybuilders/easybuild-easyconfigs/actions/runs/3053240278/jobs/4923608804#step:8:304 is 
```
AssertionError: False is not true : Patch file /home/runner/work/easybuild-easyconfigs/easybuild-easyconfigs/easybuild/easyconfigs/a/AmberTools-20_cmake-locate-netcdf.patch/AmberTools-20_cmake-locate-netcdf.patch is available for Amber-22.0-foss-2021b-AmberTools-22.3.eb
```

Reply:
Yeah, did that in the Amber PR (though I made the typo of writing just "alt_location" first... fixed now.